### PR TITLE
Make KeyPair / SecretKey methods safe

### PR DESCRIPTION
**Pull request by Geod24**
Should be a no brainer

### CODE REVIEW

**Review by hewison-chris — APPROVED** [review]
LGTM

File: source/agora/crypto/Key.d [hunk 1]
```diff
@@ -322,7 +326,7 @@ public struct SecretKey
         import std.format : phobos_format = format;
         import ocean.text.convert.Formatter : ocean_format = format;
         assert(phobos_format("%s", sd) == "**SECRET**");
-        assert(ocean_format("{}", sd) == "**SECRET**");
+        () @trusted { assert(ocean_format("{}", sd) == "**SECRET**"); }();
```
Comment (linked0) [inline source/agora/crypto/Key.d:329]: Why is the `ocean_format` nested in `@trusted`?

Comment (Geod24) [reply source/agora/crypto/Key.d:329]: Because `format` isn't `@safe`, and having the `@safe` at the top also applies to `unittest`.

Comment (linked0) [reply source/agora/crypto/Key.d:329]: Is `phobos_format` `@safe`? I can't find `@safe` in the prototype of the `format` function in `std.format`.

**Review by linked0 — APPROVED** [review]

**Pull request merged**